### PR TITLE
style: reorg tutorial metadata

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/consul/tutorials/vault-secure/vault-pki-consul-secure-tls) 🔎
- [Asana task](https://app.asana.com/0/home/1199634971443477/1204604036286170) 🎟️
- [Figma file 🎨 ](https://www.figma.com/file/NAFjIPkbwjyY8jPl6T9nlP/Tutorial-Variants?type=design&node-id=297-36193&t=PlF0QTBCVhYIMs2w-0)

## 🗒️ What

This PR restyles the tutorial metadata area in preparation for the addition of the variant picker UI, which will render below the metadata area. The layout, spacing, and color of the entire tutorial metadata area has been reworked to match[ this new spec](https://www.figma.com/file/NAFjIPkbwjyY8jPl6T9nlP/Tutorial-Variants?type=design&node-id=297-36193&t=PlF0QTBCVhYIMs2w-0). These changes are independent of variants and can be released on their own, as there will still be many tutorials without variants, so the metadata changes are global despite variant state. 

Screenshot 2023-05-15 at 5 36 28 PM

## 📸 Design Screenshots
Screenshot 2023-05-16 at 10 28 42 AM
Screenshot 2023-05-16 at 10 29 26 AM

## 🧪 Testing

- [Visit a tutorial ](https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/consul/tutorials/vault-secure/vault-pki-consul-connect-ca)with lots of metadata, interactive lab, various badge types etc 
- Ensure that it works in dark and light theme.
- Ensure it scales between mobile / desktop
- Test on all browsers
- Log in and test the authenticated state
- Bookmark and un-bookmark a tutorial to validate that the icons update as expected.
- More examples: 
- https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/nomad/tutorials/integrate-vault/vault-postgres
- https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/nomad/tutorials/autoscaler/dynamic-application-sizing
- https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/consul/tutorials/resiliency/introduction-chaos-engineering
- [Visit the library](https://dev-portal-git-ksreorg-tutorial-metadata-hashicorp.vercel.app/tutorials/library) and choose different combinations of metadata to validate that all variations work.

### CODE REVIEW

**Comment by kendallstrautman** [comment]
💭 Should separate badge work into own PR

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx [hunk 1]
```diff
@@ -18,10 +19,13 @@ export default function InteractiveLabButton() {
 
 	return (
 		<Button
+			className={s.button}
 			text={buttonText}
 			onClick={() => ctx.setActive(!ctx.active)}
 			icon={<IconTerminalScreen16 />}
 			data-heap-track="interactive-lab-button"
+			color="secondary"
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/interactive-lab-button/index.tsx:27]: Potentially not including this

File: src/components/bookmark-button/index.tsx [hunk 2]
```diff
@@ -54,16 +54,23 @@ export function BookmarkButtonIconOnly({
  * only in the tutorial meta component.
  */
 
-function BookmarkButtonTextAndIcon({
+function BookmarkButtonSecondaryIcon({
```
Comment (kendallstrautman) [inline src/components/bookmark-button/index.tsx:57]: Also potentially not including this depending on the state of feedback

File: src/components/tutorial-meta/components/badges/badges.module.css [hunk 3]
```diff
@@ -4,32 +4,29 @@
  */
 
 .list {
-	--badge-spacing: 8px;
+	--tag-spacing: 12px;
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/badges/badges.module.css:7]:
```suggestion
	--badge-spacing: 12px;
```

File: src/components/tutorial-meta/components/badges/badges.module.css [hunk 4]
```diff
@@ -4,32 +4,29 @@
...
 
 .list {
-	--badge-spacing: 8px;
+	--badge-spacing: 12px;
 
 	align-items: flex-start;
 	display: flex;
 	flex-wrap: wrap;
-	gap: var(--badge-spacing);
+	gap: var(--tag-spacing);
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/badges/badges.module.css:12]:
```suggestion
	gap: var(--badge-spacing);
```

File: src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css [hunk 5]
```diff
@@ -107,7 +107,7 @@ for further details.
 .breadcrumbContainer {
 	display: flex;
 	justify-content: flex-start;
-	margin-bottom: 32px;
+	margin-bottom: 24px;
```
Comment (kendallstrautman) [inline src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css:110]: I confirmed with @braydenlove that this update should be global to all sidebar / sidecar layouts.

File: src/components/tutorial-meta/components/interactive-lab-button/index.tsx [hunk 6]
```diff
@@ -22,6 +22,8 @@ export default function InteractiveLabButton() {
...
+			size="small"
```
Comment (kendallstrautman) [inline src/components/tutorial-meta/components/interactive-lab-button/index.tsx:26]: In the designs, this spans the entire width at <tablet sizes, but after looking at it _in situ_ we decided to have it wrap in the same way as on desktop. cc @braydenlove for the final call there though.

File: src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css (hunk 5 above)
Comment (braydenlove) [reply src/layouts/sidebar-sidecar/sidebar-sidecar-layout.module.css:110]: Thanks, also got confirmation from @heitzke that this is a good global refinement.

**Review by braydenlove — APPROVED** [review]
Awesome work, I checked this out on a variety of browsers and devices and it is feeling really good. Look forward to any possible refinements later when we look at the data!

**Review by zchsh — APPROVED** [review]
✅ Ran through testing steps, everything looks good to me 👍

---

⛏️ One minor nitpick I'm unsure about - I'm not seeing the `- (minus)` symbol show up on bookmark icon hover (on Firefox):

CleanShot 2023-05-16 at 16 25 41

**Comment by kendallstrautman** [comment]
Will push up the code comment for the a11y tradeoff once git wakes back up 😎

**Pull request merged**